assets/search/components/SearchResultsBar/SearchResultsAdvancedSearchRow.tsx

```diff
@@ -70,103 +83,38 @@ export function SearchResultsAdvancedSearchRow({
         }
     };
 
-    if (keywords.all.length) {
-        advancedSearchTags.push(
-            <Tag
-                key="tags-advanced--and"
-                text={gettext('and')}
-                operator={true}
-                shade="success"
-                readOnly={true}
-            />
-        );
-
-        keywords.all.forEach((term, index) => {
-            advancedSearchTags.push(
-                <Tag
-                    key={`tags-advanced--and-${index}`} // using index instead of term which might not be unique
-                    text={term}
-                    shade="success"
-                    readOnly={readonly}
-                    onClick={(event) => {
-                        event.preventDefault();
-                        removeKeywordEntry('all', index);
-                        refresh?.();
-                    }}
-                />
-            );
-        });
-    }
-
-    if (keywords.any.length) {
-        if (advancedSearchTags.length) {
-            advancedSearchTags.push(
-                <span
-                    key="tag-advanced--separator-any"
-                    className="tag-list__separator"
-                />
-            );
-        }
-
-        advancedSearchTags.push(
-            <Tag
-                key="tags-advanced--or"
-                text={gettext('or')}
-                operator={true}
-                shade="info"
-                readOnly={true}
-            />
-        );
-
-        keywords.any.forEach((term, index) => {
+    let key: keyof typeof keywords;
+    for (key in keywords) {
+        // due to the declaration outside key would be the last key
+        // at the end but we need the current value for the onClick callback
+        const field = key;
```

Comment:
not exactly happy about this @tomaskikutis @thecalcc are you aware of any better way?

Reply:
You're missing `const`

```
for (const key in keywords) {
     ^
```

With `const` it keeps the context and `onChange` would access correct values.

![image](https://github.com/superdesk/newsroom-core/assets/2076953/b9a2b7d3-777f-41c1-9949-372b84a524cb)

The problem then becomes typescript inferring key as string and not union type. one option is to overwrite types for global `Object.keys` by putting the following to `globals.d.ts`

```
interface ObjectConstructor {
    keys<T extends {}>(o: T): Array<keyof T>;
}
```

In case it causes stuff not to compile somewhere else, we could add a custom object keys implementation that's more strict:

```
function keysStrict<T extends {}>(obj: T): Array<keyof T> {
    return Object.keys(obj) as Array<keyof T>;
}

for (const field of keysStrict(keywords)) {
    ...
}
```

Third option, to use unmodified `Object.keys` and cast the result:
```
for (const field of Object.keys(keywords) as Array<keyof typeof keywords>) {
    ...
}
```

Reply:
went for the 3rd after all, that seems to be the most straightforward option